### PR TITLE
Macos boxes for ci

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
This PR documents how to create and manage macOS CI nodes. Because macOS is not supported by our current cloud providers, these instructions are geared towards creating VMs on physical machines we would need to host and manage ourselves, i.e. these notes are mostly targeted at @nycnewman.

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
Removed `garys-macos-pool` from Azure.

**Comment by garyverhaegen-da** [comment]
This is now ready for review.

File: infra/macos/1-create-box/README.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+where, again, this is the commit I tested with, so I am providing it for
+maximum reproducibility, but I have no reason to mistrust the master branch.
+
+Finally, run:
+```bash
+rbenv install 2.7.0
+```
+
+which installs Ruby 2.7.0 for the current user.
```
Comment (nycnewman) [inline infra/macos/1-create-box/README.md:132]: Needs to include a command to change default version to 2.7.0 or it still uses system version and you get a WritePermission error on next step

Comment (garyverhaegen-da) [reply infra/macos/1-create-box/README.md:132]: This should be done the modification to your shell init script on line 97 above. Have you tried in a new terminal?

Comment (garyverhaegen-da) [reply infra/macos/1-create-box/README.md:132]: Ah, I see. It looks like I forgot a file. Added. Thanks for pointing out!

**Comment by garyverhaegen-da** [comment]
Confirmed with @nycnewman that this seems to be working for now. We may need to tweak things more in the near future, but that will be done in separate work streams if needed.

**Pull request merged**